src/kernel/network.cpp

```diff
@@ -55,211 +152,209 @@ CryptoKernel::Network::Network(CryptoKernel::Log* log,
-                                    newSeed["lastseen"] = 0;
-                                    newSeed["height"] = 1;
-                                    newSeed["score"] = 0;
-                                    peers->put(dbTx.get(), addr.toString(), newSeed);
-                                }
-                            } else {
-                                changeScore(it->first, 10);
-                                throw Peer::NetworkError();
-                            }
-                        }
-                    } catch(const Json::Exception& e) {
-                        changeScore(it->first, 50);
-                        throw Peer::NetworkError();
-                    }
-
-                    const std::time_t result = std::time(nullptr);
-                    it->second->info["lastseen"] = static_cast<uint64_t>(result);
-                } catch(const Peer::NetworkError& e) {
-                    log->printf(LOG_LEVEL_WARN,
-                                "Network(): Failed to contact " + it->first + ", disconnecting it");
-                    removals.insert(it->first);
-                }
-            }
-
-            for(const auto& peer : removals) {
-                const auto it = connected.find(peer);
-                peers->put(dbTx.get(), peer, it->second->info);
-                if(it != connected.end()) {
-                    connected.erase(it);
-                }
-            }
-
-            for(const auto& peer : peerInfos) {
-                peers->put(dbTx.get(), peer.first, peer.second);
-            }
-
-            dbTx->commit();
+void CryptoKernel::Network::makeOutgoingConnectionsWrapper() {
+	while(running) {
+		bool wait = false;
+		makeOutgoingConnections(wait);
+		if(wait) {
+			std::this_thread::sleep_for(std::chrono::milliseconds(20));
+		}
+		else {
+			std::this_thread::sleep_for(std::chrono::milliseconds(100));
+		}
+	}
+}
 
-            std::lock_guard<std::mutex> cLock(connectedStatsMutex);
-            connectedStats.clear();
+void CryptoKernel::Network::infoOutgoingConnectionsWrapper() {
+	while(running) {
+		infoOutgoingConnections();
+		std::this_thread::sleep_for(std::chrono::milliseconds(100));
+	}
+}
 
-            for(const auto& peer : connected) {
-                peerStats stats = peer.second->peer->getPeerStats();
-                stats.version = peer.second->info["version"].asString();
-                stats.blockHeight = peer.second->info["height"].asUInt64();
-                connectedStats.insert(std::make_pair(peer.first, stats));
-            }
-        }
+void CryptoKernel::Network::makeOutgoingConnections(bool& wait) {
+	std::map<std::string, Json::Value> peersToTry;
+	std::vector<std::string> peerIps;
+
+	CryptoKernel::Storage::Table::Iterator* it = new CryptoKernel::Storage::Table::Iterator(
+			peers.get(), networkdb.get());
+
+	for(it->SeekToFirst(); it->Valid(); it->Next()) {
+		if(connected.size() >= 8) { // honestly, this is enough
+			wait = true;
+			return;
```

Comment:
This doesn't actually fix the problem because `it` is not deleted. It's actually a memory leak/the db will stay locked forever. It would be better to make the `it` into a `unique_ptr` to ensure it gets deleted if the function returns.